internal/engine/experiment.go

```diff
@@ -204,7 +207,7 @@ func (e *experiment) MeasureWithContext(ctx context.Context, input string) (*mod
 	// by adding the test keys etc. Please, note that, as of 2024-06-05, we're using
 	// the measurement Input to provide input to an experiment. We'll probably
 	// change this, when we'll have finished implementing richer input.
-	measurement := e.newMeasurement(input)
+	measurement := e.newMeasurement(target.Input())
```

Comment:
A follow-up diff will modify `newMeasurement` to also read the options from the `target`.